From 2b6c1fed8eb356c9671b840befc5e7f59f955992 Mon Sep 17 00:00:00 2001
From: Katsuya Matsubara <matsu@igel.co.jp>
Date: Thu, 14 Jun 2012 16:54:17 +0900
Subject: [PATCH 42/80] rcarvin: remove unused variable 'image_mode' in the device private structure

The boolean 'image_mode' in the device private structre could keep
whether image color format is NV16 or not.
Unfortunately the boolean must be referred only in rcar_vin_set_fmt().

Signed-off-by: Katsuya Matsubara <matsu@igel.co.jp>
---
 drivers/media/video/rcarvin.c |   13 +++++--------
 1 files changed, 5 insertions(+), 8 deletions(-)

diff --git a/drivers/media/video/rcarvin.c b/drivers/media/video/rcarvin.c
index 96f6531..6d9bdcb 100644
--- a/drivers/media/video/rcarvin.c
+++ b/drivers/media/video/rcarvin.c
@@ -156,8 +156,6 @@ struct rcar_vin_dev {
 
 	enum v4l2_field field;
 
-	unsigned int image_mode:1;
-
 	/* physical address for video capture memory */
 	dma_addr_t dma_handle;
 	/* virtual address for video capture memory */
@@ -1678,7 +1676,7 @@ static int rcar_vin_set_fmt(struct soc_camera_device *icd,
 	unsigned int vin_sub_width = 0, vin_sub_height = 0;
 	u16 scale_v, scale_h;
 	int ret;
-	bool image_mode;
+	bool can_scale;
 	enum v4l2_field field;
 
 	dev_geo(dev, "S_FMT(pix=0x%x, %ux%u)\n",
@@ -1714,16 +1712,16 @@ static int rcar_vin_set_fmt(struct soc_camera_device *icd,
 
 	switch (pixfmt) {
 	case V4L2_PIX_FMT_NV16:
-		image_mode = false;
+		can_scale = false;
 		break;
 	default:
-		image_mode = true;
+		can_scale = true;
 	}
 
 	dev_geo(dev, "4: request camera output %ux%u\n", mf.width, mf.height);
 
 	/* 5. - 9. */
-	ret = client_scale(icd, &mf, &vin_sub_width, &vin_sub_height, image_mode);
+	ret = client_scale(icd, &mf, &vin_sub_width, &vin_sub_height, can_scale);
 
 	dev_geo(dev, "5-9: client scale return %d\n", ret);
 
@@ -1752,7 +1750,7 @@ static int rcar_vin_set_fmt(struct soc_camera_device *icd,
 
 	pix->colorspace = mf.colorspace;
 
-	if (image_mode) {
+	if (can_scale) {
 		/* Scale pix->{width x height} down to width x height */
 		scale_h		= calc_scale(vin_sub_width, &pix->width);
 		scale_v		= calc_scale(vin_sub_height, &pix->height);
@@ -1776,7 +1774,6 @@ static int rcar_vin_set_fmt(struct soc_camera_device *icd,
 	icd->current_fmt	= xlate;
 
 	pcdev->field = field;
-	pcdev->image_mode = image_mode;
 
 	return 0;
 }
-- 
1.7.0.4

